spec/spec_helper.rb

```diff
@@ -1,51 +1,39 @@
+# frozen_string_literal: true
+
 require 'pathname'
 dir = Pathname.new(__FILE__).parent
 $LOAD_PATH.unshift(dir, File.join(dir, 'fixtures/modules/augeasproviders_core/spec/lib'), File.join(dir, '..', 'lib'))
 
-require 'rubygems'
+# Managed by modulesync - DO NOT EDIT
+# https://voxpupuli.org/docs/updating-files-managed-with-modulesync/
 
-require 'simplecov'
-unless RUBY_VERSION =~ /^1\.8/
-  require 'coveralls'
-  SimpleCov.formatter = Coveralls::SimpleCov::Formatter
-end
-SimpleCov.start do
-  add_group "Puppet Types", "/lib/puppet/type/"
-  add_group "Puppet Providers", "/lib/puppet/provider/"
+# puppetlabs_spec_helper will set up coverage if the env variable is set.
+# We want to do this if lib exists and it hasn't been explicitly set.
+ENV['COVERAGE'] ||= 'yes' if Dir.exist?(File.expand_path('../lib', __dir__))
 
-  add_filter "/spec/fixtures/"
-  add_filter "/spec/unit/"
-  add_filter "/spec/support/"
+RSpec.configure do |c|
+  c.mock_with :mocha
 end
 
-require 'puppetlabs_spec_helper/module_spec_helper'
+require 'voxpupuli/test/spec_helper'
 require 'augeas_spec'
 
-Puppet[:modulepath] = File.join(dir, 'fixtures', 'modules')
-
-# There's no real need to make this version dependent, but it helps find
-# regressions in Puppet
-#
-# 1. Workaround for issue #16277 where default settings aren't initialised from
-# a spec and so the libdir is never initialised (3.0.x)
-# 2. Workaround for 2.7.20 that now only loads types for the current node
-# environment (#13858) so Puppet[:modulepath] seems to get ignored
-# 3. Workaround for 3.5 where context hasn't been configured yet,
-# ticket https://tickets.puppetlabs.com/browse/MODULES-823
-#
-ver = Gem::Version.new(Puppet.version.split('-').first)
-if ver >= Gem::Version.new("2.7.20")
-    puts "augeasproviders: setting $LOAD_PATH to work around broken type autoloading"
-    Puppet.initialize_settings
-    $LOAD_PATH.unshift(
-        dir,
-        File.join(dir, 'fixtures/modules/augeasproviders_core/spec/lib'),
-        File.join(dir, 'fixtures/modules/augeasproviders_core/lib')
-    )
-
-    
-    $LOAD_PATH.unshift(File.join(dir, '..', 'lib'))
+if File.exist?(File.join(__dir__, 'default_module_facts.yml'))
+  facts = YAML.safe_load(File.read(File.join(__dir__, 'default_module_facts.yml')))
+  facts&.each do |name, value|
+    add_custom_fact name.to_sym, value
+  end
 end
 
-# Load all shared contexts and shared examples
-Dir["#{dir}/support/**/*.rb"].sort.each {|f| require f}
+ver = Gem::Version.new(Puppet.version.split('-').first)
+if ver >= Gem::Version.new('2.7.20')
```

Comment:
Not sure if you want to tackle this, but I think it's safe to assume we're beyond this now ;) I also wonder if this is still needed though.

Reply:
I cleaned it up and moved it to `spec/spec_helper_local.rb`. spec_helper.rb is now coming from modulesync, without modifications.